libs/common/src/types/mongo-db-models/tfm-facilities.ts

```diff
@@ -121,4 +121,5 @@ export type TfmFacility = {
   _id: ObjectId;
   facilitySnapshot: Facility;
   amendments?: TfmFacilityAmendment[];
+  tfm: object;
```

Comment:
Do we want to add any more details to the type or is the plan to incrementally update this?

Reply:
Would be great if we could add some types to this as the other existing types seem to be fully populated 🙏

Reply:
The code regarding this tfm object is quite hard to follow and I am not close to this at all, so felt best to leave it vague and then incrementally improve when someone with more context has time to do so. If either can explain the purpose of this field, the possible fields it can have and their types then that would be really helpful!

Reply:
For reference, adding the full `TfmFacilityAmendment` type took about a week of running e2e tests and tracking and documenting every field which was sent to the `dtfs-central-api` `PUT /v1/tfm/.../amendments` endpoint. That was technically easier than this `Facility.tfm` type though because it was very straightforward to track down that this was the only endpoint where amendments were modified

Reply:
Happy to leave for now :)

Reply:
I'm a bit rusty on this, but if I remember correctly, `facility.tfm` is modified/created during deal submission to TFM, before any amendments.

Can see where the initial data is created starting from here => https://github.com/UK-Export-Finance/dtfs2/blob/df67bba1f917bdd9f04717f990b92786e327ca97/trade-finance-manager-api/src/v1/controllers/update-facilities.js#L23

Does look fiddly to obtain the amendments data structure along with this, not a blocker